supplementary_style_guide/glossary_terms_conventions/product_conventions/red-hat-single-sign-on.adoc

```diff
@@ -0,0 +1,146 @@
+
+[discrete]
+[[credentials]]
+== Credentials
+Credentials are pieces of data that Red Hat Single Sign-On uses to verify the identity of a user. Some examples are passwords, one-time-passwords, digital certificates, or even fingerprints.
+
+[discrete]
+[[direct-grant]]
+== Direct grant
+A way for a client to obtain an access token on behalf of a user through a REST invocation.
+
+[discrete]
+[[events]]
+== Events
+Events are audit streams that admins can view and hook into.
+
+[discrete]
+[[groups]]
+== Groups
+Groups manage groups of users. Attributes can be defined for a group. You can also map roles to a group. Users that become members of a group inherit the attributes and role mappings that group defines.
+
+[discrete]
+[[identity-provider]]
+== Identity provider
+An identity provider (IDP) is a service that can authenticate a user. Red Hat Single Sign-On is an IDP.
+
+[discrete]
+[[identity-provider-federation]]
+== Identity provider federation
+Red Hat Single Sign-On can be configured to delegate authentication to one or more IDPs. Social login through Facebook or Google+ is an example of identity provider federation. You can also hook Red Hat Single Sign-On to delegate authentication to any other OpenID Connect or SAML 2.0 IDP.
+
+[discrete]
+[[identity-provider-mappers]]
+== Identity provider mappers
+When doing IDP federation you can map incoming tokens and assertions to user and session attributes. This helps you propagate identity information from the external IDP to your client requesting authentication.
+
+[discrete]
+[[identity-token]]
+== Identity token
+A token that provides identity information about the user. Part of the OpenID Connect specification.
+
+[discrete]
+[[protocol-mappers]]
+== Protocol mappers
+For each client you can tailor what claims and assertions are stored in the OIDC token or SAML assertion. You do this per client by creating and configuring protocol mappers.
+
+[discrete]
+[[realms]]
+== Realms
+A realm manages a set of users, credentials, roles, and groups. A user belongs to and logs into a realm. Realms are isolated from one another and can only manage and authenticate the users that they control.
+
+[discrete]
+[[required-actions]]
+== Required actions
+Required actions are actions a user must perform during the authentication process. A user cannot complete the authentication process until these actions are complete. For example, an admin might schedule users to reset their passwords every month. An update password required action is set for all these users.
+
+[discrete]
+[[roles]]
+== Roles
+Roles identify a type or category of user. `Admin`, `user`, `manager`, and `employee` are all typical roles that may exist in an organization. Applications often assign access and permissions to specific roles rather than individual users because dealing with users can be too granular and hard to manage.
```

Comment:
```suggestion
Identifies a type or category of user. `Admin`, `user`, `manager`, and `employee` are all typical roles that may exist in an organization. Applications often assign access and permissions to specific roles rather than individual users because dealing with users can be too granular and hard to manage.
```